engines/suppliers/app/controllers/suppliers/api/v1/suppliers_controller.rb

```diff
@@ -0,0 +1,64 @@
+module Suppliers
+  module Api
+    module V1
+      class SuppliersController < ApplicationController
+        before_action :load_supplier, only: [:show, :update, :destroy]
+
+        # GET /api/v1/suppliers
+        #
+        def index
+          suppliers = SuppliersFinder.new(supplier_params[:group_id]).find
```

Comment:
Not from personal code, there are several partially implemented examples around, all geared to have a memory-based repository implemented in a transparent way. I think that the gist of it is to try and use activerecord objects in a memory repository, and implement the Repository class so those can work (which puts in the spotlight the complex issues of persistence - much more complex than retrieval - in a generic repository you should be able to check out an object and then synchronize it back to the repo, or not)